sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ConnectionOptions.java

```diff
@@ -94,14 +90,15 @@ public ConnectionOptions(String fullyQualifiedNamespace, TokenCredential tokenCr
      *     {@code clientOptions}, {@code hostname}, or {@code verifyMode} is null.
      */
     public ConnectionOptions(String fullyQualifiedNamespace, TokenCredential tokenCredential,
-        CbsAuthorizationType authorizationType, AmqpTransportType transport, AmqpRetryOptions retryOptions,
-        ProxyOptions proxyOptions, Scheduler scheduler, ClientOptions clientOptions,
+        CbsAuthorizationType authorizationType, String authorizationScope, AmqpTransportType transport,
+        AmqpRetryOptions retryOptions, ProxyOptions proxyOptions, Scheduler scheduler, ClientOptions clientOptions,
         SslDomain.VerifyMode verifyMode, String product, String clientVersion, String hostname, int port) {
 
         this.fullyQualifiedNamespace = Objects.requireNonNull(fullyQualifiedNamespace,
             "'fullyQualifiedNamespace' is required.");
         this.tokenCredential = Objects.requireNonNull(tokenCredential, "'tokenCredential' is required.");
         this.authorizationType = Objects.requireNonNull(authorizationType, "'authorizationType' is required.");
+        this.authorizationScope = Objects.requireNonNull(authorizationScope, "'authorizationScope' is required.");
```

Comment:
Even though this is implementation, is there any requirements around backwards compatibility in case we run into a dependency diamond problem?